src/core.c/IO/Path.rakumod

```diff
@@ -650,35 +654,38 @@ my class IO::Path is Cool does IO {
     }
 
     # slurp given path with given normalized encoding
-    sub slurp-path-with-encoding(str $path, str $encoding) {
+    sub slurp-path-with-encoding(str $path, str $encoding, int $size) {
         CATCH { return $_.Failure }
 
-        nqp::elems(my $blob := slurp-path-bin($path))
+        nqp::elems(my $blob := slurp-path-bin($path, $size))
           ?? nqp::join("\n",nqp::split("\r\n",nqp::decode($blob,$encoding)))
           !! ""
     }
 
     proto method slurp() {*}
     multi method slurp(IO::Path:D: :$bin!) {
+        my $size = max try self.s, fallback-slurp-size;
```

Comment:
FYI, infix max (i.e., `self.s max fallback-slurp-size`) is quite a bit faster than sub/method max.